### PR TITLE
[#4] Add GeoDNS filtering plugin

### LINKED ISSUE #4: geodns plugin
We need to implement [geodns](https://github.com/abh/geodns) functionlity.

First we should support at least ([NNS compatible](https://github.com/nspcc-dev/neo-go/blob/e0779f2d6e7fe92ced50ae29e274e0fe50fb62dd/examples/nft-nd-nns/nns.go#L311-L318) records) and determining exact min distance server (geodns support weighted servers and more 5% minimal distance)

### PR DESCRIPTION
**Pull request by KirillovDenis**
Add geodns plugin that filters response records and preserve only nearest (less or equal than max value from config) to the client 

closes #4 

Signed-off-by: Denis Kirillov <denis@nspcc.ru>

### CODE REVIEW

**Review by alexvanin — COMMENTED** [review]
I suppose we are using MaxMind-DB data there. CC BY-SA 3.0 license requires to add an attribution note somewhere, if I understand correctly.  Maybe add readme or license txt file in testdata dir for that? 

/cc @realloc

**Comment by realloc** [comment]
We are not shipping the database itself. It's up to the user to obtain the GeoIP files via any legal channel, hence we only need to mention the required DB format and possible ways to get it.

**Comment by alexvanin** [comment]
> We are not shipping the database itself. It's up to the user to obtain the GeoIP files via any legal channel, hence we only need to mention the required DB format and possible ways to get it.

It is being used in the unit tests, so technically we ship it.

**Comment by realloc** [comment]
> It is being used in the unit tests, so technically we ship it.

Indeed we do... Let's create a tiny db in the same format with fake geo data just for test?

**Review by alexvanin — APPROVED** [review]

File: plugin/geodns/health.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+func newHealthChecker() *checker {
+	return &checker{
+		cache: gcache.New(1000).Expiration(30 * time.Second).Build(),
+		client: &http.Client{
+			Timeout: 2 * time.Second,
+		},
+		schema: "https://",
+		port:   ":80",
```
[the remark this answers is not in the document]
Comment (KirillovDenis) [reply plugin/geodns/health.go]: Fixed

**Pull request merged**